Mathlib/Order/Minimal.lean

```diff
@@ -93,6 +94,44 @@ theorem eq_of_mem_minimals (ha : a ∈ minimals r s) (hb : b ∈ s) (h : r b a)
   antisymm (ha.2 hb h) h
 #align eq_of_mem_minimals eq_of_mem_minimals
 
+theorem mem_maximals_iff : x ∈ maximals r s ↔ x ∈ s ∧ ∀ ⦃y⦄, y ∈ s → r x y → x = y := by
+  simp only [maximals, Set.mem_sep_iff, and_congr_right_iff]
+  refine' fun _ ↦ ⟨fun h y hys hxy ↦ antisymm hxy (h hys hxy), fun h y hys hxy ↦ _⟩
+  convert hxy <;> rw [h hys hxy]
+
+theorem mem_maximals_setOf_iff : x ∈ maximals r (setOf P) ↔ P x ∧ ∀ ⦃y⦄, P y → r x y → x = y :=
+  mem_maximals_iff
+
+theorem mem_minimals_iff : x ∈ minimals r s ↔ x ∈ s ∧ ∀ ⦃y⦄, y ∈ s → r y x → x = y :=
+  @mem_maximals_iff _ _ _ (IsAntisymm.swap r) _
```

Comment:
If `IsAntisymm.swap` was an instance instead of a theorem, then the proof here would be just `mem_maximals_iff` (which I double checked by adding `attribute [instance] IsAntisymm.swap` above).

I'll ask on Zulip to see if there is any reason `IsAntisymm.swap` can't be an instance.

Reply:
I'd suggest writing
```lean
theorem mem_minimals_iff : x ∈ minimals r s ↔ x ∈ s ∧ ∀ ⦃y⦄, y ∈ s → r y x → x = y :=
  have := IsAntisymm.swap r
  mem_maximals_iff
```
rather than figuring out which argument the instance goes in.

Regarding adding `IsAntisymm.swap` as an instance, maybe it's ok as a lower-priority one? There shouldn't be any issues with `IsAntisymm.swap` introducing cycles now that we have Lean 4's instance algorithm, but I'd just expect it being necessary to go through `IsAntisymm.swap` to be rare.

Reply:
FWIW: I agree about the suggested proof but the PR has already landed.